R/rd2markdown.R

```diff
@@ -230,7 +232,10 @@ rd2markdown.dots <- function(x, fragments = c(), ...) {
 #' @exportS3Method
 #' @rdname rd2markdown
 rd2markdown.TEXT <- function(x, fragments = c(), ...) {
-  gsub("\\s+", " ", gsub("\n", " ", x))
+  # We replace one new line sign with two to make sure that proper
+  # paragraphs are always applied. New line without paragraphs is not something
+  # used in Rd2 objects and thus we do not account for such situations.
+  gsub("\n{1}", "\n\n", x)
```

Comment:
The main change in this PR. Rationale explained in the description

Reply:
Just doing some quick tests, and single newlines are permitted in `Rd`s just like in markdown, so this will introduce separations between all the lines. For example, in this package we have this documentation, which uses newlines to keep the Rd width to 80 chars. 

https://github.com/Genentech/rd2markdown/blob/3d72b8d2c349e69db46fecc889c89d59436e8b21/man/documentation_to_markdown.Rd#L25-L31

Reply:
I made a couple minor tweaks in this branch:

https://github.com/Genentech/rd2markdown/tree/17-paragraph-splits-2

Notable changes
```r
rd2markdown.TEXT <- function(x, fragments = c(), ...) {
  if (grepl("^\\s*\n\\s*$", x)) return(rd2markdown.cr())
  trim_extra_newlines(x)
}

trim_extra_newlines <- function(x) {
  # non-newline whitespace
  ws <- "[^\\S\r\n]"
  re1 <- paste0(ws, "+", "(\n\\b)?")
  re2 <- paste0("(\n", ws, "*){2,}")
  gsub(re1, " ", gsub(re2, "\n\n", x, perl = TRUE), perl = TRUE)
}
```

Let me know what you think of that style. If you like it, feel free to merge the changes in with this branch.

Reply:
I just added one minor tweak to the description to fit it's purposed better :)